pandas/io/formats/csvs.py

```diff
@@ -33,27 +35,26 @@ class CSVFormatter:
     def __init__(
         self,
         obj,
-        path_or_buf=None,
-        sep=",",
-        na_rep="",
+        path_or_buf: Optional[FilePathOrBuffer[str]] = None,
+        sep: str = ",",
+        na_rep: str = "",
         float_format=None,
         cols=None,
-        header=True,
-        index=True,
+        header: Union[bool, Sequence[str]] = True,
```

Comment:
```suggestion
        header: Union[bool, Sequence[Hashable]] = True,
```

Yea the docstring says `str` but we really allow almost any hashable object as a column label, so just be mindful of that for annotations